velox/exec/Task.cpp

```diff
@@ -2433,6 +2434,14 @@ ContinueFuture Task::taskDeletionFuture() {
   return std::move(future);
 }
 
+std::string Task::printPlanWithStats(bool includeCustomStats) const {
+  if (planFragment_.planNode) {
```

Comment:
Should we check Task state and fail if task is not running or completed?

Reply:
A task state check is not required. On Prestissimo, I am using this on task completion. I would also think this is helpful on Task failure for debugging.